packages/mdc-radio/foundation.ts

```diff
@@ -25,7 +25,7 @@ import {MDCFoundation} from '@material/base/foundation';
 import {MDCRadioAdapter} from './adapter';
 import {cssClasses, strings} from './constants';
 
-class MDCRadioFoundation extends MDCFoundation<MDCRadioAdapter> {
+export class MDCRadioFoundation extends MDCFoundation<MDCRadioAdapter> {
```

Comment:
FYI this class is missing a constructor that calls super while mixing the default adapter plus the passed adapter. It appears to have always lacked it, even in JS.

Reply:
Yikes on bikes! Added.

I did a quick grep of all packages and I don't think we're missing any other constructors.

However, I noticed that `MDCTabScrollerRTL` (which is used by `MDCTabScrollerFoundation`) does not extend `MDCFoundation`, nor does it accept partial adapters in the constructor:

https://github.com/material-components/material-components-web/blob/56a065035c0f3f9e3f4a3fff3f3b445f9f1ffd3b/packages/mdc-tab-scroller/rtl-scroller.ts#L27-L32

It's already like that in `master`:

https://github.com/material-components/material-components-web/blob/70672f12ff19591c53fef6876c059c854f9cb816/packages/mdc-tab-scroller/rtl-scroller.js#L33-L38

Is that OK?

Reply:
That's not a foundation class, so N/A.

There are also a few foundation classes which extend other (non-base) foundation classes which already deal with the constructor and all share a common adapter, so it's N/A in those cases 2. The 2 cases I pointed out are the only 2 that seem problematic.